### PR TITLE
perf: add benchmark for compare function

### PR DESCRIPTION
**Pull request by FrancisMurillo**
Add benchmark for compare function.

#### ✔ Checklist:

- [x] Commit messages have been written in [Conventional Commits](https://www.conventionalcommits.org) format;
- [x] This PR has been added to [CHANGELOG.md](https://github.com/dotenv-linter/dotenv-linter/blob/master/CHANGELOG.md) (at the top of the list);
- [x] Tests for the changes have been added (for bug fixes / features);
- [x] Docs have been added / updated on the [dotenv-linter.github.io](https://github.com/dotenv-linter/dotenv-linter.github.io) (for bug fixes / features).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,3 +12,4 @@
 *.env
 *.env_*
 !/tests/fixes/fixtures/*.env
+!/benches/fixtures/*.env
```
Comment (mgrachev) [inline .gitignore:15]: Why these changes is needed?

File: benches/check.rs [hunk 2]
```diff
@@ -24,5 +24,26 @@ pub fn check_benchmark(c: &mut Criterion) {
     });
 }
 
-criterion_group!(benches, check_benchmark);
+pub fn compare_benchmark(c: &mut Criterion) {
```
Comment (mgrachev) [inline benches/check.rs:27]: Please move this benchmark function to `benches/compare.rs`.

File: benches/check.rs [hunk 3]
```diff
@@ -24,5 +24,26 @@ pub fn check_benchmark(c: &mut Criterion) {
...
+    let temp = tempdir().expect("create tempdir");
+    let path = temp.into_path();
+
+    let current_dir = env::current_dir().expect("get current dir");
+    let app = dotenv_linter::cli::new(current_dir.as_os_str());
+    let matches = app.get_matches_from(vec!["dotenv-linter", "compare", ".env", ".env.compare"]);
```
Comment (mgrachev) [inline benches/check.rs:33]: You should pass the real path of files, e.g.:
```rust
vec!["dotenv-linter", "compare", path.join(".env").to_str().expect("path to str"), path.join(".env.compare").to_str().expect("path to str")]
```

**Review by mgrachev — COMMENTED** [review]
@FrancisMurillo 👋 Thank you for your contribution 🙂

I have left some comments. Please take a look at them 👀

File: .gitignore (hunk 1 above)
Comment (FrancisMurillo) [reply .gitignore:15]: I noticed the `benches/fixtures/simple.env` as part of the original check benchmark function so I added an `benches/fixtures/compare.env`. I needed to commit it but it was blocked by `.gitignore`.

File: benches/check.rs (hunk 2 above)
Comment (FrancisMurillo) [reply benches/check.rs:27]: Will do

File: benches/check.rs (hunk 3 above)
Comment (FrancisMurillo) [reply benches/check.rs:33]: Just followed the pattern on how the `.env` arg was passed in the check function but will update.

**Review by mgrachev — APPROVED** [review]

**Comment by mgrachev** [comment]
@FrancisMurillo Thank you for your help 👍

🙏 If it’s not difficult for you, please support the project - star on GitHub ⭐️
